src/Form/Form.js

```diff
@@ -475,7 +486,8 @@ class Form {
 
     get submitButtons() {
         const selector = this.matching.cssSelector('submitButtonSelector');
-        const allButtons = /** @type {HTMLElement[]} */ (findEnclosedElements(this.form, selector));
+        const buttons = this.form.querySelectorAll(selector);
+        const allButtons = /** @type {HTMLElement[]} */ (buttons.length > 0 ? [...buttons] : findElementsInShadowTree(this.form, selector));
```

Comment:
Same as before just took out the code for checking the length of visible elements, which was being does inside the `findEnclosedElements` function.